api/sdk/sdk.go

```diff
@@ -60,58 +55,85 @@ type memberKeys struct {
 
 // SDK is used to send messages to API
 type SDK struct {
-	apiURLs    *ringBuffer
-	rootMember *requester.UserConfigJSON
-	logLevel   interface{}
+	apiURLs               *ringBuffer
+	rootMember            *requester.UserConfigJSON
+	migrationAdminMember  *requester.UserConfigJSON
+	migrationDamonMembers []*requester.UserConfigJSON
+	logLevel              interface{}
 }
 
 // NewSDK creates insSDK object
-func NewSDK(urls []string, rootMemberKeysPath string) (*SDK, error) {
+func NewSDK(urls []string, memberKeysDirPath string) (*SDK, error) {
 	buffer := &ringBuffer{urls: urls}
 
-	rawConf, err := ioutil.ReadFile(rootMemberKeysPath)
-	if err != nil {
-		return nil, errors.Wrap(err, "[ NewSDK ] can't read keys from file")
-	}
+	getMember := func(keyPath string, ref string) (*requester.UserConfigJSON, error) {
 
-	keys := memberKeys{}
-	err = json.Unmarshal(rawConf, &keys)
-	if err != nil {
-		return nil, errors.Wrap(err, "[ NewSDK ] can't unmarshal keys")
+		rawConf, err := ioutil.ReadFile(keyPath)
+		if err != nil {
+			return nil, errors.Wrap(err, "[ getMember ] can't read keys from file")
+		}
+
+		keys := memberKeys{}
+		err = json.Unmarshal(rawConf, &keys)
+		if err != nil {
+			return nil, errors.Wrap(err, "[ getMember ] can't unmarshal keys")
+		}
+
+		return requester.CreateUserConfig(ref, keys.Private)
 	}
 
 	response, err := requester.Info(buffer.next())
 	if err != nil {
 		return nil, errors.Wrap(err, "[ NewSDK ] can't get info")
 	}
 
-	rootMember, err := requester.CreateUserConfig(response.RootMember, keys.Private)
+	rootMember, err := getMember(memberKeysDirPath+"root_member_keys.json", response.RootMember)
+	if err != nil {
+		return nil, errors.Wrap(err, "[ NewSDK ] can't get root member")
+	}
+
+	migrationAdminMember, err := getMember(memberKeysDirPath+"migration_admin_member_keys.json", response.MigrationAdminMember)
 	if err != nil {
-		return nil, errors.Wrap(err, "[ NewSDK ] can't create user config")
+		return nil, errors.Wrap(err, "[ NewSDK ] can't get migration admin member")
 	}
 
-	return &SDK{
-		apiURLs:    buffer,
-		rootMember: rootMember,
-		logLevel:   nil,
-	}, nil
+	result := &SDK{
+		apiURLs:               buffer,
+		rootMember:            rootMember,
+		migrationAdminMember:  migrationAdminMember,
+		migrationDamonMembers: []*requester.UserConfigJSON{},
+		logLevel:              nil,
+	}
 
+	if len(response.MigrationDamonMembers) < insolar.GenesisAmountActiveMigrationDamonMembers {
+		return nil, errors.Wrap(err, "[ NewSDK ] need at least '"+strconv.Itoa(insolar.GenesisAmountActiveMigrationDamonMembers)+"' migration damons")
+	}
+
+	for i := 0; i < insolar.GenesisAmountActiveMigrationDamonMembers; i++ {
+		m, err := getMember(memberKeysDirPath+"migration_damon_"+strconv.Itoa(i)+"_member_keys.json", response.MigrationDamonMembers[i])
```

Comment:
You use this structure '"migration_damon_"+strconv.Itoa(i)+"_member_keys.json"' two times.
Make constant with template

Reply:
for example where?

Reply:
look for "k, err := secrets.GetPublicKeyFromFile(g.config.MembersKeysDir + "migration_damon_" + strconv.Itoa(i) + "_member_keys.json")"
on this page

Reply:
for example where should I create constant template?